engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbFormState.java

```diff
@@ -32,6 +33,8 @@ public class DbFormState implements MutableFormState {
   private final DbCompositeKey<DbString, DbLong> idAndVersionKey;
   private final ColumnFamily<DbCompositeKey<DbString, DbLong>, PersistedForm>
       formByIdAndVersionColumnFamily;
+  private final Object2ObjectHashMap<String, Object2ObjectHashMap<DirectBuffer, PersistedForm>>
```

Comment:
I prefer to use the same approach of other state classes for 2 reason:
- Better readability, same behaviour of other state classes
- The performance are not different if we use a composite key instead of nested map

Reply:
* I communicated with @remcowesterhoud if there are any reasons that we use nested maps here and he also agreed that we can use composite keys. Also, we are thinking to create refactoring issues for other classes to use composite keys.
* How performance is not different by using composite key?

Reply:
Insertion and retrieval are constant in time, also for nested map. If we use composite key we shouldn't get a boost in terms of performance. I don't like to use composite key in general, mainly because you increase the possibility to have collisions

Reply:
Insertion and retrievals are constant time in best case, worst case is `O(n)`. In that sense nested maps makes them `O(2n)` in the worst case. When you use composite keys, it is `O(n)` in the worst case. How do we say that it increases collision compared to single keys without looking at data?

As I mentioned above, using composite keys is expected to **relatively** increase retrieval and insertion performance from `O(2n)` to `O(n)` in the worst case. I think, it is definitely a better approach, but surely not a blocker in our case.

We can also ask for input from the team. WDYT?

Reply:
I was talking about a general approach, not this specific case.